03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,28 +1,68 @@
 const Task = require('../models/Task');
+const asyncWrapper = require('../middleware/async');
+const { createCustomError } = require('../errors/custom-errors');
 
-const getAllTasks = (req, res) => {
-  res.send('get all tasks from the file');
-};
+//we pass req, res to asyncWrapper
+const getAllTasks = asyncWrapper(async (req, res) => {
+  const tasks = await Task.find({});
+
+  //Options for response:
+  res.status(200).json({ tasks });
+  //or tasks: tasks
+
+  // res.status(200).json({ tasks, amount : tasks.length })
+
+  //  res.status(200).json({ success: true, data: {tasks, numberOfHits : tasks.length} })
 
-const createTask = async (req, res) => {
+  //res
+  // .status(200)
+  // .json({ status: "success", data: {tasks, numberOfHits : tasks.length} })
+  //with this las vestion we should change line 11 in public/browser.app.js : data: { data: {tasks} },
+});
+
+const createTask = asyncWrapper(async (req, res) => {
   const task = await Task.create(req.body);
   res.status(201).json({ task });
-  //server sends us back  the body with the data provided by user in json format () in this case "name": "shake",
-  //"completed": true
-};
+});
 
-const getTask = (req, res) => {
-  //server sends us back the body with the id provided by user in json format
-  res.json({ id: req.params.id });
-};
+//server sends us back  the body with the data (provided by user) in json format () in this case "name": "shake",
+//"completed": true...etc.
```

Comment:
just double-checking; this part of the comment is referring to what happens with `Task.create` up on line 24 right?

It seems like it got shifted a bit in this PR and might be unclear if you come back to read this again further in the futrue :slightly\_smiling\_face:

Reply:
absolutely, this comment is for the line 24. It looks messy but I find it really helpful to leave explicit comments :)